pkg/clients/clients.go

```diff
@@ -161,25 +167,52 @@ func (dcs *DynamicClientSet) gvkForKind(kind Kind) (gvk schema.GroupVersionKind,
 	return
 }
 
-func (dcs *DynamicClientSet) namespaced(gvk schema.GroupVersionKind) bool {
-	resources, err := dcs.DiscoveryClientCached.ServerPreferredResources()
+func (dcs *DynamicClientSet) namespaced(gvk schema.GroupVersionKind) (bool, error) {
+	// Handle known non-namespaced Kinds.
+	switch Kind(gvk.Kind) {
+	case APIService, CertificateSigningRequest, ClusterRole, ClusterRoleBinding, CustomResourceDefinition,
+		MutatingWebhookConfiguration, Namespace, PersistentVolume, PodSecurityPolicy, PriorityClass,
+		StorageClass, ValidatingWebhookConfiguration:
```

Comment:
lol I don't like this but I also want to not block the review. Can we add a comment explaining why we have this lookup table? I understand this to be because the API server can be flaky, which is ???? scary?

Reply:
I think the flakiness comes from the discovery client. It requires the caller to handle retries when `Invalidate()` is called, and won't return resources while the cache is refreshing. I'll add a comment and TODO to see if there's a cleaner way to handle that.